### PR TITLE
Add Location object and ToText methods

### LINKED ISSUE #42: Location object to contain file name and span
Create a Location object which will contain a file name/path and the span of the error for locations where errors occur.

Update `RunChecks` to provide the file name to the Error (and create method to take in a location).

Create a convert for Location to string (@awakeman to provide the format wanted for Location string) and a convert for Error to string (@awakeman to do same for that).

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #42 

 ### Test files

 ### Changelog

 ### Additional comments

### CODE REVIEW

File: Test_Engine/Compute/RunChecks.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static ComplianceResult RunChecks(this SyntaxNode node)
                         Create.ComplianceResult(
                             errLevel == ErrorLevel.Error ? ResultStatus.CriticalFail : ResultStatus.Fail,
                             new List<Error> {
-                                Create.Error(message,result,errLevel,method.Name)
+                                Create.Error(message, Create.Location(path, result.ToLineSpan("")), errLevel, method.Name)
```
Comment (awakeman) [inline Test_Engine/Compute/RunChecks.cs]: To convert to a `LineSpan` you need the contents of the file

File: Test_Engine/Query/Checks/ContainsCopyright.cs [hunk 2]
```diff
@@ -37,7 +37,7 @@ public static partial class Query
         public static Span ContainsCopyright(CompilationUnitSyntax node)
         {
             Error error = Test.Query.ContainsCopyright(node.GetLeadingTrivia(), -1).Errors.FirstOrDefault();
-            if (error != null) return error.Location;
+            if (error != null) return error.Location.Line.ToSpan("");
```
Comment (awakeman) [inline Test_Engine/Query/Checks/ContainsCopyright.cs]: As above

Comment (FraserGreenroyd) [reply Test_Engine/Query/Checks/ContainsCopyright.cs]: Done

**Comment by FraserGreenroyd** [comment]
Both changes made ready for re-review

**Comment by FraserGreenroyd** [comment]
Typo fixed in PR title for change log - ready for re-review

**Pull request merged**